File: Java/Easy/Accepted/baseball-game_3.java
class Solution {
    public int calPoints(String[] operations) {
        
        Stack<Integer> stack = new Stack<Integer>();
        int sum = 0;
        for(int i = 0; i < operations.length; i++){
            if(operations[i].equals("C")){
                sum -= stack.pop();
            }else if(operations[i].equals("D")){
                int temp = stack.peek() * 2;
                sum += temp;
                stack.push(temp);
            }else if(operations[i].equals("+")){
                int temp = stack.pop();
                int temp2 = stack.peek() + temp;
                sum += temp2;
                stack.push(temp);
                stack.push(temp2);
            }else{
                int temp = Integer.parseInt(operations[i]);
                sum += temp;
                stack.push(temp);
            }
        }
        return sum;
    }
}

File: Rust/Easy/Accepted/baseball-game_3.rs
impl Solution {
    pub fn cal_points(operations: Vec<String>) -> i32 {
        
        let mut scores: Vec<i32> = Vec::new();
        for op in operations {
            match op.as_str() {
                "C" => {
                    scores.pop();
                }
                "D" => {
                    if let Some(&score) = scores.last() {
                        scores.push(score * 2);
                    }
                }
                "+" => {
                    if let Some(&score1) = scores.last() {
                        if let Some(&score2) = scores.get(scores.len() - 2) {
                            scores.push(score1 + score2);
                        }
                    }
                }
                _ => {
                    scores.push(op.parse::<i32>().unwrap());
                }
            }
        }
        scores.iter().sum()
    }
}